src/index/IndexImpl.h

```diff
@@ -347,20 +347,37 @@ class IndexImpl {
   // --------------------------------------------------------------------------
   // TEXT RETRIEVAL
   // --------------------------------------------------------------------------
-  std::string_view wordIdToString(WordIndex wordIndex) const;
+  struct TextBlockMetadataAndWordInfo {
+    TextBlockMetaData tbmd_;
+    bool hasToBeFiltered_;
+    IdRange<WordVocabIndex> idRange_;
+  };
```

Comment:
Add some comments as for the semantics, 
is this some metadata used for retrieving a single block?
And is it correct, that we need the `idRange_` only when `hasToBeFiltered_` is true?
Then it can be `optional<IdRange>` with a comment, the `hasToBeFiltered` can be a member function, and the computation of `hasToBeFiltered` can be inside this class (currently it is somewhere else, where it is confusing and mixed together with other things).

But maybe we best keep this for a separate PR, make sure to not it down.

Reply:
(Add a TODO here for now describing these ideas in short s.t. we remember them.